longhaultests/src/iothub_longhaul.ts

```diff
@@ -0,0 +1,128 @@
+#!/usr/bin/env node
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+'use strict';
+import * as uuid from 'uuid';
+import * as async from 'async';
+import * as dbg from 'debug';
+const debug = dbg('longhaul:main');
+
+import { Registry, ConnectionString as HubConnectionString } from 'azure-iothub';
+import { ConnectionString as DeviceConnectionString } from 'azure-iot-device';
+
+import { D2CSender } from './d2c_sender';
+
+const MAX_EXECUTION_TIME = parseInt(process.env.MAX_EXECUTION_TIME_SECONDS) * 1000;
+const SEND_INTERVAL = parseInt(process.env.DEVICE_SEND_INTERVAL_SECONDS) * 1000;
+const SEND_TIMEOUT = parseInt(process.env.DEVICE_SEND_TIMEOUT_SECONDS) * 1000;
+
+const MAX_CREATE_TIME = 30000;
+const MAX_DELETE_TIME = 30000;
+const MAX_START_TIME = 30000;
+const MAX_STOP_TIME = 30000;
+
+const ERROR_EXIT_CODE = -1;
+const OK_EXIT_CODE = 0;
+
+const hubConnectionString = process.env.IOTHUB_CONNECTION_STRING;
+const registry = Registry.fromConnectionString(hubConnectionString);
+
+const deviceId = 'node-longhaul-sak-' + uuid.v4();
+let protocol;
```

Comment:
>p [](start = 4, length = 1)

Nit: this holds a constructor function, so it should be Protocol with a capital P.  (tslint has flagged this for me before, but you're not calling new() directly so you probably wouldn't see it).

Reply:
in that case tslint will bark about it being uppercase because in this context, we're not calling it with new, and tslint rules dictate variable names to be camel cased.

---
In reply to: [177507571](https://github.com/Azure/azure-iot-sdk-node/pull/257#discussion_r177507571) [](ancestors = 177507571)